sdk/storage/Azure.Storage.Files/src/Models/NtfsAttributes.cs

```diff
@@ -0,0 +1,137 @@
+        /// </summary>
+        Directory = FileAttributes.Directory,
+
+        /// <summary>
+        /// The file is a candidate for backup or removal.
+        /// </summary>
+        Archive = FileAttributes.Archive,
+
+        /// <summary>
+        /// The file or directory is temporary. A temporary file contains data that is needed while an
+        /// application is executing but is not needed after the application is finished.
+        /// File systems try to keep all the data in memory for quicker access rather than
+        /// flushing the data back to mass storage. A temporary file should be deleted by
+        /// the application as soon as it is no longer needed.
+        /// </summary>
+        Temporary = FileAttributes.Temporary,
+
+        /// <summary>
+        /// The file or directory is offline. The data of the file is not immediately available.
+        /// </summary>
+        Offline = FileAttributes.Offline,
+
+        /// <summary>
+        /// The file or directory will not be indexed by the operating system's content indexing service.
+        /// </summary>
+        NotContentIndexed = FileAttributes.NotContentIndexed,
+
+        /// <summary>
+        /// The file or directory is excluded from the data integrity scan. When this value
+        /// is applied to a directory, by default, all new files and subdirectories within
+        /// that directory are excluded from data integrity.
+        /// </summary>
+        NoScrubData = FileAttributes.NoScrubData
+    }
+
+    /// <summary>
+    /// CloudFileNtfsAttributesHelper helper.
+    /// </summary>
+    internal class CloudFileNtfsAttributesHelper
+    {
+        private static readonly Dictionary<NtfsAttributes, string> dictionary = new Dictionary<NtfsAttributes, string>()
+        {
+            { NtfsAttributes.ReadOnly, "ReadOnly" },
+            { NtfsAttributes.Hidden, "Hidden" },
+            { NtfsAttributes.System, "System" },
+            { NtfsAttributes.Normal, "Normal" },
+            { NtfsAttributes.Directory, "Directory" },
+            { NtfsAttributes.Archive, "Archive" },
+            { NtfsAttributes.Temporary, "Temporary" },
+            { NtfsAttributes.Offline, "Offline" },
+            { NtfsAttributes.NotContentIndexed, "NotContentIndexed" },
+            { NtfsAttributes.NoScrubData, "NoScrubData" },
+        };
+
+        /// <summary>
+        /// Converts a CloudFileNtfsAttributes to a string
+        /// </summary>
+        /// <param name="attributes"><see cref="NtfsAttributes"/></param>
+        /// <returns>string</returns>
+        internal static string ToString(NtfsAttributes? attributes)
+            => attributes == null
+                ? null
+                : String.Join("|", dictionary.Select(
+                    r =>
+                    {
+                        return attributes.Value.HasFlag(r.Key) ? r.Value : null;
+                    })
+                .Where(r => r != null));
```

Comment:
This works, but it's slower and uses more memory than a `StringBuilder` and a bunch of `if` checks.

Reply:
fixed.